File: src/matlab/Alunos_Aula2/Easom.m
function Y = Easom (X)

x1 = X(1);
x2 = X(2);
    
  m=2;  % Dimension
  
   Y=-cos(x1)*cos(x2)*exp(-(x1-pi)^2-(x2-pi)^2); 
    
end

File: ene300/functions/_easom.py
import numpy as np
from ._function_counter import function_counter

@function_counter
def easom(x):
    # Reference
    # Global Optimization Test Problems. Retrieved June 2013, from http://www-optima.amp.i.kyoto-u.ac.jp/member/student/hedar/Hedar_files/TestGO.htm. 
    # EASOM FUNCTION Funções para testes de Algoritmos
    return -np.cos(x[0])*np.cos(x[1])*np.exp(-(x[0]-np.pi)**2-(x[1]-np.pi)**2)
